### ISSUE TITLE
include displayName in docs

### ISSUE THREAD

**Issue by iamdustan** [reactjs/react-docgen#30]
Is there a reason this wasn’t already included? Having a displayName with each description would be beneficial for tools that are using docgen without having to re-read the js file to detect the display name again.

This wasn’t that big of a deal with `React.createClass` or `class extends React.Component` since those are pretty limited call sites, but the stateless function components can get a bit gnarly in trying to infer their display names. The detection for these are already pretty solid in the #28 branch.

**Comment by fkling** [comment]
You mean why it isn't listed in the README? Probably just an oversight from my side.

**Comment by iamdustan** [comment]
wait. does the displayName already appear in there?

**Comment by iamdustan** [comment]
I don’t think it does... this is the result of console.logging

```
docgen.parse(fs.readFileSync('./my-component.js'))
```

```
{ description: '@private',
  props:
   { to: { type: [Object], required: true, description: '' },
     name: { type: [Object], required: true, description: '' } } }
```

**Comment by fkling** [comment]
We do have https://github.com/reactjs/react-docgen/blob/master/src/handlers/displayNameHandler.js though.

**Comment by fkling** [comment]
I can't repro this btw. The `displayName` shows up in the result. What does your `./my-component.js` look like?

**Comment by iamdustan** [comment]
Hmm..

so I have stateless components not getting a `displayName` as well as

```
import React, {PropTypes, Component} from 'react';
class Tabs extends Component {
  ....
}

export default Tabs;
```

**Comment by fkling** [comment]
Oh, the `displayName` handler only handles explicit `displayName` properties. You want it to also return the inferred name from the class or function name, right?

**Comment by iamdustan** [comment]
:+1:

...does babel still follow that approach for displayName handling?

**Comment by mik01aj** [comment]
@fkling, yes! (I know the question wasn't to me, but I'd love to have it this way :))

**Comment by fkling** [comment]
I don't know, will have to look into this. Seems reasonable to me though. PRs are also welcone ;)

**Comment by iamdustan** [comment]
I vote @mik01aj for those one :)

**Comment by AlanFoster** [comment]
Babel will generate the display name for you when it is not supplied explicitly. 

:+1: to implementing a similar inferrance rule to react-docgen

**Comment by AlanFoster** [comment]
@iamdustan Would this be something that is feasible to implement?

**Comment by mik01aj** [comment]
Maybe you could just use this plugin: https://github.com/gajus/babel-plugin-react-display-name

**Comment by NogsMPLS** [comment]
I would love for react-docgen to be able to infer the name of a component based on the Component.name.

Babel only creates a displayName for the ```.createClass()``` convention, not for the es6 classes or stateless functional components.

I'll dig around react-docgen today and see if I can figure out a PR that might accomplish this.
